hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java

```diff
@@ -110,17 +117,32 @@ public void enrichEvent(StructuredTrace trace, Event event) {
    * enricher class itself could be renamed to be more specific.
    */
   private void enrichHostHeader(Event event) {
-    for (String key: HOST_HEADER_ATTRIBUTES) {
-      String value = SpanAttributeUtils.getStringAttribute(event, key);
-      if (value != null && !value.isEmpty()) {
-        // Ignore if it's "localhost"
-        String host = sanitizeHostValue(value);
-        if (!LOCALHOST.equalsIgnoreCase(host)) {
-          addEnrichedAttribute(event, HOST_HEADER, AttributeValueCreator.create(host));
-          break;
+    Protocol protocol = EnrichedSpanUtils.getProtocol(event);
+    if (protocol == Protocol.PROTOCOL_GRPC) {
+      if (event.getGrpc() != null && event.getGrpc().getRequest() != null && event.getGrpc().getRequest().getRequestMetadata() != null) {
+        String value = event.getGrpc().getRequest().getRequestMetadata().getAuthority();
+        if (sanitizeAndEnrichHostHeader(event, value)) {
```

Comment:
I agree that the function name is not clear. But I don't think we should move enrichment out of the code path as that would lead to duplicated code.

Reply:
Renamed method to `isHostHeaderEnrichmentSuccessful`

Reply:
I don't see multiple invocations of `addEnrichedAttribute(event, HOST_HEADER, AttributeValueCreator.create(host));` as duplicated code. But, that could have been avoided as well by creating a new method `enrichHostHeader` which just calls the `addEnrichedAttribute` inside

Reply:
`isHostHeaderEnrichmentSuccessful` -> is also a misleading name. That means the enrichment on the span failed due to some reasons, instead of saying that enrich host header with valid host header

Reply:
created a method `getSanitizedHostHeader`